www/include/monitoring/objectDetails/xml/serviceSendCommand.php

```diff
@@ -60,47 +62,66 @@
 $pearDB = new CentreonDB();
 $hostObj = new CentreonHost($pearDB);
 $svcObj = new CentreonService($pearDB);
-$host_id = $_GET["host_id"];
-$svc_id = $_GET["service_id"];
-$poller = $hostObj->getHostPollerId($host_id);
-$cmd = $_GET["cmd"];
-$sid = session_id();
-$act_type = $_GET["actiontype"];
+
+$hostId = filter_var(
+    $_GET['host_id'] ?? false,
+    FILTER_VALIDATE_INT
+);
+
+$serviceId = filter_var(
+    $_GET['service_id'] ?? false,
+    FILTER_VALIDATE_INT
+);
+
+$pollerId = $hostObj->getHostPollerId($hostId);
+
+$cmd = $_GET['cmd'];
+$actionType = $_GET['actiontype'];
 
 $pearDB = new CentreonDB();
 
-$DBRESULT = $pearDB->query("SELECT session_id FROM session WHERE session.session_id = '" . $sid . "'");
-if (!$DBRESULT->rowCount()) {
+if ($sessionId = session_id()) {
+    $res = $pearDB->prepare("SELECT * FROM `session` WHERE `session_id` = :sid");
+    $res->bindValue(':sid', $sessionId, PDO::PARAM_STR);
+    $res->execute();
+    if (!$session = $res->fetch(PDO::FETCH_ASSOC)) {
+        exit();
+    }
+} else {
     exit();
 }
 
-if ($centreon->user->is_admin() === 0) {
+/* If admin variable equals 1 it means that user admin
+ * otherwise it means that it is a simple user under ACL
+ */
+$isAdmin = (int) $centreon->user->access->admin;
+if ($centreon->user->access->admin === 0) {
     if (!$centreon->user->access->checkAction($cmd)) {
         exit();
     }
-    if (!$centreon->user->access->checkHost($host_id)) {
+    if (!$centreon->user->access->checkHost($hostId)) {
         exit();
     }
-    if (!$centreon->user->access->checkService($svc_id)) {
+    if (!$centreon->user->access->checkService($serviceId)) {
         exit();
     }
 }
 
 
 $command = new CentreonExternalCommand($centreon);
-$cmd_list = $command->getExternalCommandList();
+$commandList = $command->getExternalCommandList();
 
-$send_cmd = $cmd_list[$cmd][$act_type];
+$sendCommand = $commandList[$cmd][$actionType];
 
-$send_cmd .= ";" . $hostObj->getHostName($host_id) . ";" . $svcObj->getServiceDesc($svc_id) . ";" . time();
-$command->setProcessCommand($send_cmd, $poller);
-$act_type ? $return_type = 1 : $return_type = 0;
+$sendCommand .= ";" . $hostObj->getHostName($hostId) . ";" . $svcObj->getServiceDesc($serviceId) . ";" . time();
```

Comment:
if serviceId === false, then getServiceDesc will try to return this on line 127 : 
```
        if (isset($svcTab[bool false])) {
            return $svcTab[bool false];
        } 
```

Reply:
lindly add a null check on the getServiceDesc() method

Reply:
**Done** Thanks :+1: